### PR TITLE
rationalize `CondTools/BeamSpot` and add unit tests to the package

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

Title says it all:
   * plugins now recognized by the framework (before they weren't due to a mistake in the `BuildFile.xml`)
   * reduce massive code duplication in `CondTools/BeamSpot/plugins` thanks to templated functions.
   * migrate to `eSConsumes`
   * implement `fillDescription` method
   * introduce unit tests

#### PR validation:

Relies on the newly introduced unit tests.

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

N/A

### CODE REVIEW

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by tvami** [comment]
Hi @mmusich there is one more ES get in this package:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-6a297c/19382/llvm-analysis/report-ddca36.html#EndPath

Can you maybe include that in this PR, too? Thanks!

**Comment by mmusich** [comment]
@tvami, 

> Can you maybe include that in this PR, too? Thanks!

taken care of in commit a4b9d06

**Comment by mmusich** [comment]
please test

**Comment by mmusich** [comment]
SA looks happy now: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-6a297c/19403/summary.html

**Comment by tvami** [comment]
+alca
 * very clean results

**Comment by tvami** [comment]
+db

File: CondTools/BeamSpot/plugins/BeamSpotOnlineRecordsWriter.cc [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+
+//
+// constructors and destructor
+//
+BeamSpotOnlineRecordsWriter::BeamSpotOnlineRecordsWriter(const edm::ParameterSet& iConfig) {
+  //now do what ever initialization is needed
+  fIsHLT = iConfig.getParameter<bool>("isHLT");
+  fasciiFileName = iConfig.getUntrackedParameter<std::string>("InputFileName");
+  fasciiFile.open(fasciiFileName.c_str());
+  if (iConfig.exists("IOVStartRun") && iConfig.exists("IOVStartLumi")) {
```
Comment (perrotta) [inline CondTools/BeamSpot/plugins/BeamSpotOnlineRecordsWriter.cc:74]: "IOVStartRun" and "InputFileName" parameters are defined in the `fillDescriptions` method, and therefore they will always exist.
As such, `fuseNewSince` can never be false: either the code can be simplified, or there is a flaw in the logic.

**Comment by mmusich** [comment]
please test

**Comment by mmusich** [comment]
@tvami @perrotta ping

**Comment by tvami** [comment]
+db

**Pull request merged**